[Python][Flaky Test] test_sync_pubsub_combined_exact_and_pattern_one_client - callback count mismatch

### Test Name

TestSyncPubSub::test_sync_pubsub_combined_exact_and_pattern_one_client[SubscriptionMethod.Blocking-MessageReadMethod.Callback-True]

### Test Location

python/tests/sync_tests/test_sync_pubsub.py:837 (assertion at tests/utils/pubsub_test_utils.py:1137)

### Failure Permlink

https://github.com/valkey-io/valkey-glide/actions/runs/26735171689/job/78786853761#logs

### Frequency

Observed in scheduled CI run on 2026-06-01. Failed in 1 out of 2 Python PubSub test jobs.

### Steps to Reproduce

1. Run the FMT Python workflow on schedule
2. Observe failure in Python PubSubTests - 3.9, EngineVersion - 6.2, Target - aarch64-unknown-linux-gnu

### System Information

GitHub Actions (Ubuntu aarch64)

### Language and Version

Python 3.9

### Engine Version

redis 6.2

### Logs

```
tests/sync_tests/test_sync_pubsub.py::TestSyncPubSub::test_sync_pubsub_combined_exact_and_pattern_one_client[SubscriptionMethod.Blocking-MessageReadMethod.Callback-True] FAILED [ 82%]

            # check that we received all messages
            assert all_channels_and_messages == {}
    
>           sync_check_no_messages_left(
                method, listening_client, callback_messages, NUM_CHANNELS * 2
            )

tests/sync_tests/test_sync_pubsub.py:837: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

method = <MessageReadMethod.Callback: 2>
client = <glide_sync.glide_client.GlideClusterClient object at 0xf05136332670>
callback = [PubSubMsg(message=b'EiOOLkoUTj', channel=b'{channel}:jdvDi', pattern=None), PubSubMsg(message=b'CkkhnJxCWV', channel=...nel=b'{channel}:uPRPI', pattern=None), PubSubMsg(message=b'uBoZtVwlPI', channel=b'{channel}:YKYkg', pattern=None), ...]
expected_callback_messages_count = 512

    def sync_check_no_messages_left(
        method: MessageReadMethod,
        client,
        callback: Optional[List[Any]] = None,
        expected_callback_messages_count: int = 0,
    ) -> None:
        """
        Verify there are no more messages to read (sync version).
    
        Args:
            method: The read method being used
            client: The sync client to check
            callback: Callback message list (for Callback method)
            expected_callback_messages_count: Expected number of messages in callback list
    
        Raises:
            AssertionError if there are unexpected messages
        """
        import pytest
    
        from tests.utils.utils import run_sync_func_with_timeout_in_thread
    
        if method == MessageReadMethod.Async:
            # assert there are no messages to read
            with pytest.raises(TimeoutError):
                run_sync_func_with_timeout_in_thread(
                    lambda: client.get_pubsub_message(),  # This blocks indefinitely
                    timeout=3.0,
                )
        elif method == MessageReadMethod.Sync:
            assert client.try_get_pubsub_message() is None
        else:
            assert callback is not None
>           assert len(callback) == expected_callback_messages_count
E           AssertionError

tests/utils/pubsub_test_utils.py:1137: AssertionError

=========================== short test summary info ============================
FAILED tests/sync_tests/test_sync_pubsub.py::TestSyncPubSub::test_sync_pubsub_combined_exact_and_pattern_one_client[SubscriptionMethod.Blocking-MessageReadMethod.Callback-True] - AssertionError
= 1 failed, 843 passed, 738 skipped, 8015 deselected, 17 warnings in 4343.36s (1:12:23) =
```

### Screenshots

_No response_

### Glide Version

main

### Expected Behavior

All expected callback messages (512 = NUM_CHANNELS * 2) should be received before the assertion check.

### Actual Behavior

The callback message count does not match the expected count of 512. This is a known recurring issue — previously tracked in #5167 (closed not_planned), #4933, #4862, and #4364 (all closed). The sync test versions were not updated with the fix from PR #4880.

### Possible Fixes

Apply the same fix from PR #4880 (making channel names unique per index) to the sync test versions. See #5167 for context.

--- a/python/tests/sync_tests/test_sync_pubsub.py
+++ b/python/tests/sync_tests/test_sync_pubsub.py
@@ -773,16 +773,17 @@ class TestSyncPubSub:
         """
 
         NUM_CHANNELS = 256
-        PATTERN = "{{{}}}:{}".format("pattern", "*")
+        PATTERN = "{pattern}:*"
 
-        # Create dictionaries of channels and their corresponding messages
+        # Create dictionaries of channels and their corresponding messages.
+        # Deterministic, index-unique names keep exact channels and the
+        # subscribed pattern from overlapping, so callback counts stay exact.
         exact_channels_and_messages = {
-            "{{{}}}:{}".format("channel", get_random_string(5)): get_random_string(10)
-            for _ in range(NUM_CHANNELS)
+            f"{{channel}}:exact_{i}": f"exact_message_{i}" for i in range(NUM_CHANNELS)
         }
         pattern_channels_and_messages = {
-            "{{{}}}:{}".format("pattern", get_random_string(5)): get_random_string(5)
-            for _ in range(NUM_CHANNELS)
+            f"{{pattern}}:match_{i}": f"pattern_message_{i}"
+            for i in range(NUM_CHANNELS)
         }
 
         all_channels_and_messages = {

